frontend/src/components/Map.tsx

```diff
@@ -22,17 +22,14 @@ const Map = ({
   currentLocation,
   events,
 }: MapProps) => {
-  const navigation = useNavigation<any>();
+  const { navigateTo } = useNavigationContext();
 
   const openEventDetails = useCallback(
     (index: number) => {
       if (showInfo) return;
-      navigation.navigate("EventDetails", {
-        id: events?.[index].id,
-        map: false,
-      });
+      navigateTo({ page: "EventDetails", id: events ? events[index].id : '' });
```

Comment:
You need to pass map: false here. Other than that this is good to merge

Reply:
Oh I didn't add it here because undefined will just evaluate to false

Reply:
I set it up inside of the EventDetails page to evaluate undefined as True lol since I didn't want to define true everywhere since its more common. This is the only instance where its false